Add option to select geometry source

pjhartzell opened the pull request:
**Related Issue(s):**
#48

**Description:**
- Added abiliity to create Item geometry from a raster data footprint. 
- The method for creating Item geometry is now a required argument in the `create_item` function. There are 4 options.

Tests and mypy will be cleaned up once we firm up the approach.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [ ] Code lints properly (run `scripts/lint`).
- [ ] Tests pass (run `scripts/test`).
- [ ] Documentation has been updated to reflect changes, if applicable.
- [ ] Examples have been updated to reflect changes, if applicable
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).

pjhartzell committed:
add option to select geometry source

gadomski commented on src/stactools/landsat/constants.py:
```diff
@@ -9,6 +9,13 @@ class Sensor(Enum):
     OLI_TIRS = "C"
 
 
+class GeometrySource(Enum):
+    USGS = "usgs"
+    FOOTPRINT = "footprint"
+    ANG = "ang"
+    BBOX = "bbox"
```
I see that these enum choices are documented in the `create_item` docstring. I think either
1. That documentation should be moved to these enum choices, or
2. This enum's docstring should point the user to `create_item` for more information about what the choices mean

pjhartzell replied:
Implemented #2.

gadomski commented on src/stactools/landsat/utils.py:
```diff
@@ -134,3 +149,69 @@ def recursive_round(coordinates: List[Any], precision: int) -> List[Any]:
     return item
+
+
+def update_geometry(
```
```suggestion
def _update_geometry(
```

Is there any reason to add this function to the public API? If not, should be made "private", as it's got a pretty complex signature.

pjhartzell replied:
Nah. Underscore added.

pjhartzell committed:
make update_item function private; point to create_item docstring

pjhartzell committed:
just a pin until the opportunity to complete this feature

- should be limited to to revising all tests for the new create_item
function signature
